openedx/core/djangoapps/user_authn/api/form_fields.py

```diff
@@ -163,3 +211,129 @@ def add_job_title_field():
     # user to input the Job Title
     job_title_label = _("Job Title")
     return _add_field_with_configurable_select_options('job_title', job_title_label)
+
+
+def add_first_name_field():
+    """
+    Returns the first name field description
+    """
+    # Translators: This label appears above a field which allows the
+    # user to input the First Name
+    first_name_label = _("First Name")
+
+    return {
+        'name': 'first_name',
+        'type': SUPPORTED_FIELDS_TYPES['TEXT'],
+        'label': first_name_label,
+        'error_message': accounts.REQUIRED_FIELD_FIRST_NAME_MSG,
+    }
+
+
+def add_last_name_field():
+    """
+    Returns the last name field description
+    """
+    # Translators: This label appears above a field which allows the
+    # user to input the Last Name
+    last_name_label = _("Last Name")
+
+    return {
+        'name': 'last_name',
+        'type': SUPPORTED_FIELDS_TYPES['TEXT'],
+        'label': last_name_label,
+        'error_message': accounts.REQUIRED_FIELD_LAST_NAME_MSG,
+    }
+
+
+def add_mailing_address_field():
+    """
+    Returns the mailing address field description
+    """
+    # Translators: This label appears above a field
+    # meant to hold the user's mailing address.
+    mailing_address_label = _("Mailing address")
+
+    return {
+        'name': 'mailing_address',
+        'type': SUPPORTED_FIELDS_TYPES['TEXTAREA'],
+        'label': mailing_address_label,
+        'error_message': accounts.REQUIRED_FIELD_MAILING_ADDRESS_MSG,
+    }
+
+
+def add_state_field():
+    """
+    Returns a State/Province/Region field to a description
+    """
+    # Translators: This label appears above a field
+    # which allows the user to input the State/Province/Region in which they live.
+    state_label = _("State/Province/Region")
+
+    return {
+        'name': 'state',
+        'type': SUPPORTED_FIELDS_TYPES['TEXT'],
+        'label': state_label,
+        'error_message': accounts.REQUIRED_FIELD_STATE_MSG,
+    }
+
+
+def add_city_field():
+    """
+    Returns a city field to a description
+    """
+    # Translators: This label appears above a field
+    # which allows the user to input the city in which they live.
+    city_label = _("City")
+
+    return {
+        'name': 'city',
+        'type': SUPPORTED_FIELDS_TYPES['TEXT'],
+        'label': city_label,
+        'error_message': accounts.REQUIRED_FIELD_CITY_MSG,
+    }
+
+
+def add_honor_code_field(separate_honor_and_tos=False):
+    """
+    Returns a honor code field to a description
```

Comment:
update docstring to include information that this field will be displayed directly on frontend